### PR TITLE
some baserom dumps

### PR DESCRIPTION
**Pull request by itskobold**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,9 @@ Thumbs.db
 build/
 .DS_Store
 *.ddump
+data2new.s
+data2old.s
+dump baserom msys code.txt
+*.lnk
+baserom.rar
+datadump/
```
Comment (Diegoisawesome) [inline .gitignore]: There's no need to add any of this to the gitignore for the whole project.

File: data/battle_frontier_1.s [hunk 2]
```diff
@@ -20,10 +41,30 @@ gUnknown_0860CE84:: @ 860CE84
 	.incbin "baserom.gba", 0x60ce84, 0x10
 
 gUnknown_0860CE94:: @ 860CE94
-	.incbin "baserom.gba", 0x60ce94, 0x20
+	window_template 0x00, 0x00, 0x03, 0x08, 0x10, 0x0f, 0x0010
+	window_template 0x00, 0x16, 0x03, 0x08, 0x10, 0x0f, 0x0090
+	window_template 0x00, 0x08, 0x01, 0x0e, 0x02, 0x0f, 0x0110
+	window_template 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0000
```
Comment (Diegoisawesome) [inline data/battle_frontier_1.s]: 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0000 should be replaced with window_template_terminator.

File: data/battle_frontier_1.s [hunk 3]
```diff
@@ -11,7 +11,28 @@ gUnknown_0860C988:: @ 860C988
 	.incbin "baserom.gba", 0x60c988, 0x269
 
 gUnknown_0860CBF1:: @ 860CBF1
-	.incbin "baserom.gba", 0x60cbf1, 0x283
+@ replacing .incbin "baserom.gba", 0x0060cbf1, 0x283
```
Comment (Diegoisawesome) [inline data/battle_frontier_1.s]: Remove the dump headers (@ replacing .incbin etc)

File: data/battle_frontier_1.s [hunk 4]
```diff
@@ -11,7 +11,28 @@ gUnknown_0860C988:: @ 860C988
...
+.byte 0x07, 0x01, 0x08, 0x10, 0x07, 0x01, 0x08, 0x10, 0x07, 0x01, 0x08, 0x10, 0x07, 0x01, 0x08, 0x10, 0x07, 0x01, 0x08, 0x10, 0x00, 0x02, 0x09, 0x10, 0x00, 0x02, 0x09, 0x10, 0x00, 0x02, 0x09, 0x10
```
Comment (Diegoisawesome) [inline data/battle_frontier_1.s]: Indent dumped data with a tab

File: data/battle_frontier_1.s [hunk 5]
```diff
@@ -47,124 +88,377 @@ gUnknown_0860D068:: @ 860D068
...
 
 gUnknown_0860D1C0:: @ 860D1C0
-	.incbin "baserom.gba", 0x60d1c0, 0x10
+@ replacing .incbin "baserom.gba", 0x0060d1c0, 0x10
+.byte 0x00, 0x0f, 0x08, 0x07, 0x03, 0x0c, 0x0b, 0x04, 0x01, 0x0e, 0x09, 0x06, 0x02, 0x0d, 0x0a, 0x05
 
 gUnknown_0860D1D0:: @ 860D1D0
-	.incbin "baserom.gba", 0x60d1d0, 0x44
+@ replacing .incbin "baserom.gba", 0x0060d1d0, 0x44
+.4byte 0x82c3188
```
Comment (Diegoisawesome) [inline data/battle_frontier_1.s]: Don't leave pointers in dumped data, ever. Always make sure they point to a symbol (that's kind of the reason dumping is so important).

File: data/naming_screen.s [hunk 6]
```diff
@@ -7,10 +7,16 @@ gUnknown_0858BD78:: @ 858BD78
...
+.4byte 0x8273216
+.4byte 0x8273256
+.4byte 0x8273296
+.4byte 0x82732d9
 
 gUnknown_0858BDC8:: @ 858BDC8
-	.incbin "baserom.gba", 0x58bdc8, 0x38
+@ replacing .incbin "baserom.gba", 0x0058bdc8, 0x38
+.string "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!$"
+.string "  "
```
Comment (Diegoisawesome) [inline data/naming_screen.s]: Use alignment and not an empty string. (.align 2)

File: data/mystery_event_msg.s [hunk 7]
```diff
@@ -3,32 +3,100 @@
...
+gUnknown_086757F4:: @ 86757F4
+.incbin "baserom.gba", 0x006757f4, 0x6d
+@ replacing .incbin "baserom.gba", 0x00675861, 0x78
+
+gText_MysteryGiftOldSeaMap::
+.string "Thank you for using the MYSTERY\nGIFT System.\pLet me confirm--you are {PLAYER}?\pWe received this OLD SEA MAP\naddressed to you.$"
+.string "It appears to be for use at the\nLILYCOVE CITY port.\pWhy not give it a try and see what\nit is about?$"
+.string "Thank you for using the MYSTERY\nGIFT System.$"
+.string "Oh, I’m sorry, {PLAYER}.\nYour BAG’s KEY ITEMS POCKET is full.\pPlease store something on your PC,\nthen come back for this.$"
+.string "  "
```
Comment (Diegoisawesome) [inline data/mystery_event_msg.s]: Use alignment and not an empty string. (.align 2)

**Comment by marijnvdwerf** [comment]
Regarding the pointer lists. The first sample is the current state, with numbers. Below is the version with the offsets replaced by pointers.

```asm
gUnknown_0832B6F8:: @ 832B6F8
    .4byte 0x85ef7da
    .4byte 0x85ef7ff
    .4byte 0x85ef81a
    .4byte 0x85ef846
```

```asm
gUnknown_0832B6F8:: @ 832B6F8
    .4byte gUnknown_085EF7DA
    .4byte gUnknown_085EF7FF
    .4byte gUnknown_085EF81A
    .4byte gUnknown_085EF846
```

Wasn't sure if everything was dumped already. Most pointers seem to point to strings.
https://gist.github.com/marijnvdwerf/13956c4613bc756ea28d8ed827e90947

**Pull request closed without merging**